src/applications/disability-benefits/2346/containers/OrderTable.jsx

```diff
@@ -0,0 +1,186 @@
+        };
+      });
+    })();
+
+    const UITableHeaders = Object.keys(UIDataObjs[0]).map(UIDataObjKey => {
+      switch (UIDataObjKey) {
+        case 'dateProcessed':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Date Processed
+            </th>
+          );
+        case 'product':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Model
+            </th>
+          );
+        case 'id':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Serial Number
+            </th>
+          );
+        case 'orderIntakeMethod':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Order Method
+            </th>
+          );
+        case 'checkbox':
+          return (
+            <th
+              className="form2346 vads-c-table__th--narrow vads-u-text-align--center"
+              key={UIDataObjKey}
+            >
+              Check to Order Batteries
+            </th>
+          );
+        default:
+      }
+      return UIDataObjKey;
+    });
+
+    const UITableRowsAndData = UIDataObjs.map(UIDataObj => {
+      const UIDataObjValues = Object.values(UIDataObj);
+
+      return (
+        <tr key={UIDataObj.id}>
+          {UIDataObjValues.map(UIDataObjValue => (
+            <td key={UIDataObjValue}>{UIDataObjValue}</td>
+          ))}
+        </tr>
+      );
+    });
+
+    return {
+      UITableHeaders,
+      UITableRowsAndData,
+    };
+  }
+
+  render() {
+    const { firstName, lastName, data, address, city, state, zip } = this.props;
+    let dlcTableHeaders;
+    let dlcTableRowsAndData;
+    // NOTE: Only run the following code when the DLC's data
+    // has been received to improve performance
+    if (data.length > 0) {
+      const { UITableHeaders, UITableRowsAndData } = this.createDlcUIData(data);
+      dlcTableHeaders = UITableHeaders;
+      dlcTableRowsAndData = UITableRowsAndData;
+    }
+    return (
+      <div className="vads-c-form2346">
+        {data.length > 0 && (
+          <>
+            <h2>
+              <strong>Name:</strong> {firstName} {lastName}
+            </h2>
+            <h2>
+              <strong>Address:</strong> {address} <br /> {city} {state} {zip}{' '}
+            </h2>
+            <table className="form2346 vads-c-table">
+              <tbody>
+                <tr>{dlcTableHeaders}</tr>
+                {dlcTableRowsAndData}
+              </tbody>
+            </table>
+            <button
+              type="button"
+              className="usa-button-primary va-button-primary"
+            >
+              Submit Your Order
+            </button>
+          </>
+        )}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  data: state.form2346Reducer.data == null ? '' : state.form2346Reducer.data,
```

Comment:
When we do comparisons, we almost always should use `===` over `==`(https://www.google.com/search?q=%3D%3D+vs+%3D%3D%3D+javascript). Also, it's usually not common to put ternaries inside `mapStateToProps`. Alternatively, we could either handle null values inside of our component or you could make use of `state?.reducer?.data?`, which does null checking for you 🙂

Reply:
@kelsonic  Can you give more clarification?

Reply:
@mr0sari0 Of course! What would you like more clarification on? 🙂

Reply:
> @mojaray2k I am a little confused. All of the fixes are marked resolved, yet I do not see any of these changes made in any of the branches related to this work.

He and I are pushing fixes now but we had some clarification questions that I am commenting on.

Reply:
@kelson  Your comment [here](https://github.com/department-of-veterans-affairs/vets-website/pull/11532#discussion_r369688125) is what I was looking for!